packages/origin-ui/src/__tests__/DemandForm.test.tsx

```diff
@@ -0,0 +1,249 @@
+
+jest.mock('@energyweb/market', () => {
+    return {
+        Demand: {
+            createDemand(offChainProps: any) {
+                return createDemand(offChainProps);
+            }
+        }
+    };
+});
+
+describe('DemandForm', () => {
+    afterAll(() => {
+        jest.unmock('@energyweb/market');
+    });
+
+    it.only('creating demand works', async () => {
+        const history = createMemoryHistory();
+
+        const sagaMiddleware = createSagaMiddleware();
+
+        const middleware = applyMiddleware(routerMiddleware(history), sagaMiddleware);
+
+        const store = createStore(createRootReducer(history), middleware);
+
+        Object.keys(sagas).forEach((saga: keyof typeof sagas) => {
+            sagaMiddleware.run(sagas[saga]);
+        });
+
+        store.dispatch(
+            addUser(({
+                id: '1',
+                isRole: () => true
+            } as Partial<User.Entity>) as User.Entity)
+        );
+
+        store.dispatch(updateCurrentUserId('1'));
+
+        const rendered = mount(
+            <MuiPickersUtilsProvider utils={MomentUtils}>
+                <Provider store={store}>
+                    <ConnectedRouter history={history}>
+                        <DemandForm />
+                    </ConnectedRouter>
+                </Provider>
+            </MuiPickersUtilsProvider>
+        );
+
+        const refresh = async () => {
+            await flushPromises();
+            rendered.update();
+        };
+
+        function fillInputField(name: string, value: string) {
+            const input = rendered.find(`${dataTestSelector(name)} input`);
+            const inputField = input.getDOMNode();
+
+            expect(inputField.getAttribute('name')).toBe(name);
+
+            input.simulate('change', { target: { value, name } });
+        }
+
+        const now = moment();
+
+        async function fillDate(name: string, dayOfMonth: number) {
+            rendered.find(`div${dataTestSelector(name)}`).simulate('click');
+
+            expect(document.querySelector('.MuiPickersToolbar-toolbar').textContent).toBe(
+                now.format('YYYYddd, MMM D')
+            );
+
+            const daysElements = document.querySelectorAll(
+                '.MuiPickersDay-day:not(.MuiPickersDay-hidden)'
+            );
+            (daysElements.item(dayOfMonth - 1) as HTMLElement).click();
+
+            await refresh();
+
+            expect(
+                (rendered.find(`${dataTestSelector(name)} input`).getDOMNode() as HTMLInputElement)
+                    .value
+            ).toBe(
+                now
+                    .clone()
+                    .set('date', dayOfMonth)
+                    .format('MMMM Do')
+            );
+
+            // Close Datepicker (click outside)
+            (document.querySelector('body > [role="presentation"] > div') as HTMLElement).click();
+
+            await refresh();
+
+            await waitForConditionAndAssert(
```

Comment:
Can we use some oldschool js events like onLoad?

Reply:
Maybe it would work, there's MutationObserver that could trigger events, I'm not sure if js-dom and enzyme support it. I'll experiment with it in future.